File: LeetCode_ProblemSet/Problem349.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        nums2 = {num: 1 for num in nums2}
        intersection = {}
        for num in nums1:
            if num in nums2:
                intersection[num] = True
        return [num for num in intersection]

File: LeetCode_ProblemSet/Go/Problem349.go
package main

func intersection(nums1 []int, nums2 []int) []int {
    nums := map[int]bool{}
	commons := map[int]bool {}
	ans := []int{}
	for _, num := range nums2 {
		nums[num] = true
	}
	for _, num := range nums1 {
		if _, ok := nums[num]; ok {
			commons[num] = true
		}
	}
	for num := range commons {
		ans = append(ans, num)
	}
	return ans
}